src/cli/CommandsLogic.ts

```diff
@@ -0,0 +1,252 @@
+import { HttpProvider, TransactionReceipt } from 'web3-core'
+import { merge } from 'lodash'
+
+import { sleep } from '../common/utils'
+
+// compiled folder populated by "prepublish"
+import StakeManager from './compiled/StakeManager.json'
+import RelayHub from './compiled/RelayHub.json'
+import Penalizer from './compiled/Penalizer.json'
+import Paymaster from './compiled/TestPaymasterEverythingAccepted.json'
+import Forwarder from './compiled/TrustedForwarder.json'
+
+import { Address } from '../relayclient/types/Aliases'
+import ContractInteractor from '../relayclient/ContractInteractor'
+import { GSNConfig } from '../relayclient/GSNConfigurator'
+import HttpClient from '../relayclient/HttpClient'
+import HttpWrapper from '../relayclient/HttpWrapper'
+
+interface CompiledContract {
+  abi: any
+  bytecode: string
+}
+
+interface RegisterOptions {
+  from: Address
+  stake: string | BN
+  funds: string | BN
+  relayUrl: string
+  unstakeDelay: string
+}
+
+export interface DeploymentResult {
+  relayHubAddress: Address
+  stakeManagerAddress: Address
+  penalizerAddress: Address
+  forwarderAddress: Address
+  paymasterAddress: Address
+}
+
+interface RegistrationResult {
+  success: boolean
+  transactions?: string[]
+  error?: string
+}
+
+export default class CommandsLogic {
+  private readonly contractInteractor: ContractInteractor
+  private readonly httpClient: HttpClient
+  private readonly config: GSNConfig
+  private readonly web3: Web3
+
+  constructor (host: string, config: GSNConfig, mnemonic?: string) {
+    let provider: HttpProvider | HDWalletProvider = new Web3.providers.HttpProvider(host)
+    if (mnemonic != null) {
+      // web3 defines provider type quite narrowly
+      provider = new HDWalletProvider(mnemonic, provider) as unknown as HttpProvider
+    }
+    this.contractInteractor = new ContractInteractor(provider, config)
+    this.httpClient = new HttpClient(new HttpWrapper(), config)
+    this.config = config
+    this.web3 = new Web3(provider)
+  }
+
+  async findWealthyAccount (requiredBalance = ether('2')): Promise<string | undefined> {
+    try {
+      const accounts = await this.web3.eth.getAccounts()
+      for (const account of accounts) {
+        const balance = new BN(await this.web3.eth.getBalance(account))
+        if (balance.gte(requiredBalance)) {
+          return account
+        }
+      }
+    } catch (error) {
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      throw Error(`Failed to retrieve accounts and balances: ${error}`)
+    }
+  }
+
+  async isRelayReady (relayUrl: string): Promise<boolean> {
+    const response = await this.httpClient.getPingResponse(relayUrl)
+    return response.Ready
+  }
+
+  async waitForRelay (relayUrl: string): Promise<void> {
+    const timeout = 30
+    console.error(`Will wait up to ${timeout}s for the relay to be ready`)
+
+    for (let i = 0; i < timeout; ++i) {
+      await sleep(1000)
```

Comment:
you always wait 1 second, regardless if the relay is already up.

Reply:
Fixed.